compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -532,6 +532,20 @@ impl<'a> AstValidator<'a> {
         }
     }
 
+    /// An item in `extern { ... }` cannot use non-ascii identifier.
+    fn check_foreign_item_ascii_only(&self, ident: Ident) {
+        let symbol_str = ident.as_str();
+        if !symbol_str.is_ascii() {
+            self.err_handler()
+                .struct_span_err(
+                    ident.span,
+                    "items in `extern` blocks cannot use non-ascii identifiers",
```

Comment:
Filed https://github.com/rust-lang/rust/issues/83942

should be sufficient to add a note: "This limitation may be lifted in the future; see https://github.com/rust-lang/rust/issues/83942"

Reply:
Done